Index: cpufreqd-2.3.3/src/cpufreqd_acpi.c
===================================================================
--- cpufreqd-2.3.3.orig/src/cpufreqd_acpi.c	2008-07-27 16:41:40.000000000 +0900
+++ cpufreqd-2.3.3/src/cpufreqd_acpi.c	2008-10-05 11:29:19.542579811 +0900
@@ -246,8 +246,8 @@ int find_class_device(const char *clsnam
 
 	/* read `clsname` devices */
 	devs = sysfs_get_class_devices(cls);
-	if (!cls) {
-		clog(LOG_INFO, "class '%s' not found (%s)\n", clsname,
+	if (!devs) {
+		clog(LOG_INFO, "class device '%s' not found (%s)\n", clsname,
 				strerror(errno));
 		sysfs_close_class(cls);
 		return -1;
